crates/bevy_ui/src/layout_components.rs

```diff
@@ -0,0 +1,313 @@
+
+    use super::*;
+
+    /// A query for all of the components in a [`FlexboxLayoutBundle`].
+    ///
+    /// See [`FlexboxLayoutChanged`] when attempting to use this as a query filter.
+    #[derive(WorldQuery)]
+    pub struct FlexboxLayoutQuery {
+        /// The layout algorithm used
+        pub layout_strategy: &'static LayoutStrategy,
+        /// The position of this UI node
+        pub position: &'static UiPosition,
+        /// Whether the node should be absolute or relatively positioned
+        pub position_type: &'static PositionType,
+        /// The constraints on the size of this node
+        pub size_constraints: &'static SizeConstraints,
+        /// The margin, padding and border of the UI node
+        pub spacing: &'static Spacing,
+        /// The flexbox layout parameters
+        pub flexbox_layout: &'static FlexboxLayout,
+        /// The direction of the text
+        pub text_direction: &'static TextDirection,
+        /// The behavior in case the node overflows its allocated space
+        pub overflow: &'static Overflow,
+    }
+
+    /// A type alias for when any of the components in the [`FlexboxLayoutBundle`] has been changed
+    ///
+    /// See [`FlexboxLayoutQuery`] for the data-fetching equivalent.
+    pub type FlexboxLayoutChanged = Or<(
+        Changed<LayoutStrategy>,
+        Changed<PositionType>,
+        Changed<SizeConstraints>,
+        Changed<Spacing>,
+        Changed<FlexboxLayout>,
+        Changed<TextDirection>,
+        Changed<Overflow>,
+    )>;
+
+    /// The flexbox-specific layout configuration of a UI node
+    ///
+    /// This follows the web spec closely,
+    /// you can use [guides](https://css-tricks.com/snippets/css/a-guide-to-flexbox/) for additional documentation.
+    #[derive(Component, Serialize, Deserialize, Reflect, Debug, PartialEq, Clone, Copy)]
+    #[reflect_value(PartialEq, Serialize, Deserialize)]
+    pub struct FlexboxLayout {
+        /// How items are ordered inside a flexbox
+        ///
+        /// Sets the main and cross-axis: if this is a "row" variant, the main axis will be rows.
+        pub flex_direction: FlexDirection,
+        /// Aligns this container's contents according to the cross-axis
+        pub align_items: AlignItems,
+        /// Overrides the inherited [`AlignItems`] value for this node
+        pub align_self: AlignSelf,
+        /// Aligns this containers lines according to the cross-axis
+        pub align_content: AlignContent,
+        /// Aligns this containers items along the main-axis
+        pub justify_content: JustifyContent,
+        /// Controls how the content wraps
+        pub flex_wrap: FlexWrap,
+        /// Defines how much a flexbox item should grow if there's space available
+        pub flex_grow: f32,
+        /// How to shrink if there's not enough space available
+        pub flex_shrink: f32,
+        /// The initial size of the item
+        pub flex_basis: Val,
```

Comment:
Nit: Since these are all defined in the `FlexboxLayout`, couldn't we remove the `flex_` prefix?

* `flex_wrap` -> `wrap`
* `flex_basis` -> `basis`
* `flex_direction` -> `direction`
* etc.